tests/Integration/Behaviour/Features/Context/Domain/CustomerFeatureContext.php

```diff
@@ -152,8 +155,105 @@ public function createCustomerUsingCommand(string $customerReference, TableNode
             (isset($data['birthday']) ? $data['birthday'] : null)
         );
 
+        if (Configuration::get('PS_B2B_ENABLE')) {
+            $command->setCompanyName($data['companyName']);
+        }
+
         /** @var CustomerId $id */
         $id = $commandBus->handle($command);
         SharedStorage::getStorage()->set($customerReference, $id->getValue());
     }
+
+    /**
+     * @Transform table:firstName,lastName,email,birthday,companyName
+     *
+     * @param TableNode $tableNode
+     *
+     * @return array
+     */
+    public function transformCustomers(TableNode $customersTable): array
+    {
+        return $customersTable->getHash();
+    }
+
+    /**
+     * @When I search for the phrases :searchPhrases I should get the following results:
+     *
+     * @param string $searchPhrases
+     * @param array $expectedCustomers
+     */
+    public function assertFoundCustomers(string $searchPhrases, array $expectedCustomers): void
+    {
+        $foundCustomers = $this->getQueryBus()->handle(new SearchCustomers(explode(' ', $searchPhrases)));
+
+        foreach ($expectedCustomers as $currentExpectedCustomer) {
+            $wasCurrentExpectedCustomerFound = false;
+            foreach ($foundCustomers as $currentFoundCustomer) {
+                if ($currentExpectedCustomer['email'] === $currentFoundCustomer['email']) {
+                    $wasCurrentExpectedCustomerFound = true;
+
+                    Assert::assertEquals(
+                        $currentExpectedCustomer['firstName'],
+                        $currentFoundCustomer['firstname'],
+                        sprintf(
+                            'Expected and found customers\'s first names don\'t match (%s and %s)',
+                            $currentExpectedCustomer['firstName'],
+                            $currentFoundCustomer['firstname']
+                        )
+                    );
+
+                    Assert::assertEquals(
+                        $currentExpectedCustomer['lastName'],
+                        $currentFoundCustomer['lastname'],
+                        sprintf(
+                            'Expected and found customers\'s last names don\'t match (%s and %s)',
+                            $currentExpectedCustomer['lastName'],
+                            $currentFoundCustomer['lastname']
+                        )
+                    );
+
+                    Assert::assertEquals(
+                        $currentExpectedCustomer['birthday'],
+                        $currentFoundCustomer['birthday'],
+                        sprintf(
+                            'Expected and found customers\'s birthdays don\'t match (%s and %s)',
+                            $currentExpectedCustomer['birthday'],
+                            $currentFoundCustomer['birthday']
+                        )
+                    );
+
+                    if (Configuration::get('PS_B2B_ENABLE')) {
```

Comment:
```suggestion
                    if (Configuration::get('PS_B2B_ENABLE')) {
```
Can be a variable loaded outside loops

Reply:
Maybe add a condition to be sure the company/company Name are not loaded when we are not in B2B mode?

Reply:
Not 100% sure this is what you expect, but I tried!